pkg/common/cloudconfig/azure/azure_test.go

```diff
@@ -0,0 +1,260 @@
+			filePath:  "./test/azure_config_invalid.yaml",
+			expectErr: true,
+		},
+		"succeeded to load config": {
+			filePath: "./test/azure_config_valid.yaml",
+			expectedConfig: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud:     "AzurePublicCloud",
+					TenantID:  "00000000-0000-0000-0000-000000000000",
+					UserAgent: "fleet-net-controller-manager",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "",
+					AADClientID:                 "",
+					AADClientSecret:             "",
+				},
+				Location:       "eastus",
+				SubscriptionID: "00000000-0000-0000-0000-000000000000",
+				ResourceGroup:  "test-rg",
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			config, err := LoadCloudConfigFromFile(test.filePath)
+			if test.expectErr {
+				if err == nil {
+					t.Errorf("failed to test LoadCloudConfigFromFile: expected error, got nil")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("failed to test LoadCloudConfigFromFile: expected no error, got %v", err)
+				}
+				if !reflect.DeepEqual(config, test.expectedConfig) {
+					t.Errorf("config parsed does not match, expected: %v, got: %v", test.expectedConfig, config)
+				}
+			}
+		})
+	}
+}
+
+func TestTrimSpace(t *testing.T) {
+	t.Run("test spaces are trimmed", func(t *testing.T) {
+		config := CloudConfig{
+			ARMClientConfig: azclient.ARMClientConfig{
+				Cloud:     "  test  \n",
+				UserAgent: "  test  \n",
+				TenantID:  "  test  \t \n",
+			},
+			AzureAuthConfig: azclient.AzureAuthConfig{
+				UserAssignedIdentityID:      "  test  \n",
+				UseManagedIdentityExtension: true,
+				AADClientID:                 "\n  test  \n",
+				AADClientSecret:             "  test  \n",
+			},
+			Location:       "  test  \n",
+			SubscriptionID: "  test  \n",
+			ResourceGroup:  "\r\n  test  \n",
+		}
+
+		expected := CloudConfig{
+			ARMClientConfig: azclient.ARMClientConfig{
+				Cloud:     "test",
+				TenantID:  "test",
+				UserAgent: "test",
+			},
+			Location:       "test",
+			SubscriptionID: "test",
+			ResourceGroup:  "test",
+			AzureAuthConfig: azclient.AzureAuthConfig{
+				UseManagedIdentityExtension: true,
+				UserAssignedIdentityID:      "test",
+				AADClientID:                 "test",
+				AADClientSecret:             "test",
+			},
+		}
+		config.trimSpace()
+		if config != expected {
+			t.Errorf("failed to test TrimSpace: expect config fields are trimmed, got: %v, expected: %v", config, expected)
+		}
+	})
+}
+
+func TestDefaultAndValidate(t *testing.T) {
+	tests := map[string]struct {
+		Cloud                       string
+		Location                    string
+		SubscriptionID              string
+		ResourceGroup               string
+		UseManagedIdentityExtension bool
+		UserAssignedIdentityID      string
+		AADClientID                 string
+		AADClientSecret             string
+		UserAgent                   string
```

Comment:
instead of putting all the fields here, you can use cloudConfig as input and output type.

For example
map[string] struct {
        input CloudConfig
        want  CloudConfig
        wantError bool
}

Reply:
updated.